glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/MetricsPingScheduler.kt

```diff
@@ -62,6 +62,24 @@ internal class MetricsPingScheduler(
         }
     }
 
+    /**
+     * Safely transforms a `Date` to a `String`.
+     *
+     * This is only useful for working around a crash bug on Android 8 devices.
+     */
+    internal fun safeDateToString(date: Date): String
+    {
+        // Work around a bug in Android 8 devices (SDK=25). See this platform issue
+        // https://issuetracker.google.com/issues/110848122 . Unfortunately there isn't much
+        // we could do, but since this is just a log message, we can simply add a placeholder
+        // value.
+        return try {
+            date.time.toString()
+        } catch (e: AssertionError) {
+            "<buggy Android 8>"
```

Comment:
I wanted to complain that this is a bit to simple, but I guess it's only used in logging, so we don't actually care.

Reply:
Heheh yeah, logging only :)